src/dune_rules/jsoo_rules.ml

```diff
@@ -72,6 +72,53 @@ end = struct
       | name, false -> [ "--disable"; name ])
 end
 
+module Version = struct
+  type t = int list
+
+  let of_string s : t =
+    let s =
+      match
+        String.findi s ~f:(function
+          | '+' | '-' | '~' -> true
+          | _ -> false)
+      with
+      | None -> s
+      | Some i -> String.take s i
+    in
+    String.split s ~on:'.' |> List.map ~f:int_of_string
+
+  let rec compare v v' =
+    match (v, v') with
+    | [ x ], [ y ] -> Int.compare x y
+    | [], [] -> Ordering.Eq
+    | [], y :: _ -> Int.compare 0 y
+    | x :: _, [] -> Int.compare x 0
+    | x :: xs, y :: ys -> (
+      match Int.compare x y with
+      | Eq -> compare xs ys
+      | n -> n)
+
+  let impl_version bin =
+    let open Memo.O in
+    let* _ = Build_system.build_file bin in
+    Memo.of_reproducible_fiber
+    @@ Process.run_capture_line Process.Strict bin [ "--version" ]
```

Comment:
Much better would be to have a `jsoo --config` command that would determine if this flag is supported or not. Version checking doesn't really work for development builds.

Reply:
I don't see the issue relying on `--version`. Here is the current ouput.

```
$ dune exec js_of_ocaml -- --version
5.0.1+git-5.0.1-57-g2b96ea001e-dirty
```

Also, a `--config` flag would not be available in old version of jsoo.